pkg/ccl/changefeedccl/changefeed_test.go

```diff
@@ -4519,15 +4519,25 @@ func TestChangefeedBackfillCheckpoint(t *testing.T) {
 		g := ctxgroup.WithContext(context.Background())
 		g.Go(func() error {
 			for {
-				_, err := foo.Next()
+				m, err := foo.Next()
 				if err != nil {
 					return err
 				}
+
+				if m.Resolved != nil {
+					ts := extractResolvedTimestamp(t, m)
+					if ts.IsEmpty() {
+						return errors.New("unexpected epoch resolved event")
+					}
+				}
 			}
 		})
+
 		defer func() {
 			closeFeed(t, foo)
-			_ = g.Wait()
+			if err := g.Wait(); err != nil {
+				require.Truef(t, jobs.HasErrJobCanceled(err), "err=%v", err)
+			}
```

Comment:
Did we have any instances where the job wasn't cancelled?

Reply:
You see above I added return `return errors.New("unexpected epoch resolved event")` -- so, normally, 
the job IS cancelled (that's what closeFeed does); but I want to make sure that that's the only error I allow,
and catch the above error if it happens.